src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6481,11 +6574,26 @@ syntax switch
 
                         (ParameterSymbol? parameter, TypeWithAnnotations parameterType, FlowAnalysisAnnotations parameterAnnotations, bool isExpandedParamsArgument) =
                             GetCorrespondingParameter(i, parametersOpt, argsToParamsOpt, expanded);
+
                         if (parameter is null)
                         {
-                            // If this assert fails, we are missing necessary info to visit the
-                            // conversion of a target typed construct.
-                            Debug.Assert(!IsTargetTypedExpression(argumentNoConversion) || _targetTypedAnalysisCompletionOpt?.ContainsKey(argumentNoConversion) is true);
+                            if (IsTargetTypedExpression(argumentNoConversion) && _targetTypedAnalysisCompletionOpt?.ContainsKey(argumentNoConversion) is true)
+                            {
+                                if (method is ErrorMethodSymbol)
+                                {
+                                    // The parameter matching logic above is not as flexible as the one we use in `Binder.BuildArgumentsForErrorRecovery`
+                                    // so we may end up with a pending conversion completion for an argument apparently without a corresponding parameter.
+                                    // We flush the completion with a plausible/dummy type and remove it.
+                                    TargetTypedAnalysisCompletion[argumentNoConversion](TypeWithAnnotations.Create(argument.Type));
+                                    TargetTypedAnalysisCompletion.Remove(argumentNoConversion);
```

Comment:
Consider using `TargetTypedAnalysisCompletion.Remove(argumentNoConversion, out var completion)` to remove the completion and get the value.

Reply:
Didn't find a `Remove` with two parameters. So will use the latter suggestion instead.